catkin_ws/src/planner/src/substates/utility/controller.py

```diff
@@ -47,6 +48,10 @@ def __init__(self, header_time):
         self.servers.append(self.SuperimposerServer)
         self.SuperimposerServer.wait_for_server()
 
+        self.QuaternionStateServer = actionlib.SimpleActionClient('quaternion_server', QuaternionAction)
```

Comment:
quaternion_server is a misleading name for a server that does position + quaternion

Reply:
if we're merging it in it should have the same name StateServer as before, what happens inside shouldn't be reflected to planner

Reply:
Yeah its a bad name we can change it. But its unrealistic to think we're gonna replace the old stuff all at once. We need to merge temp code so we can test it before we can deprecate everything. I think we should merge it now and clean it up later. There isn't much point optimizing code that's still going to change

Reply:
> 

The only advantage of merging right now that I see is to make it easier to test quaternions, and maybe during complex rotations (maybe?) avoiding Gimbal lock. The disadvantages are that we would have unstable and probably buggy code in the planner on the main branch that would make it much harder to test missions in the sim. I don't think it's a good idea to use the main branch as a temp merging branch, we should do that on a seperate branch no?